fmn/sse/sse_webserver.py

```diff
@@ -29,16 +29,16 @@ class SSEServer(resource.Resource):
     subscribers = SSESubscriber(log=logger)
 
     def render_GET(self, request):
-        request = self.add_headers(request)
-        request.write(b"")
         if self.is_valid_path(request.postpath):
+            request = self.add_headers(request)
             self.handle_request(request)
             request.notifyFinish().addBoth(self.responseFailed,
                                            request, request.postpath)
             return server.NOT_DONE_YET
         else:
+            request.setResponseCode(404)
             return self.invalid_request(request=request,
```

Comment:
I noticed this sets the status reason as well so the status line comes back ``HTTP/1.1 404 Invalid Path``. The normal 404 status line is ``HTTP/1.1 404 Not Found`` and I think anything else is surprising to people. A simpler way to do this would be to use the Twisted class for 404 which sets the response code, status, etc. and renders a small HTML response indicating the problem:

```
response = resource.NoResource()
return response.render(request)
```